### DISCUSSION TITLE
Programmatic createPage quietly?

### DISCUSSION THREAD

**Discussion by tonnyavery** [gatsbyjs/gatsby#36912] [Help] [answered]
I am using `onCreatePage` callback to create some extra aliases for my localized pages.
Problem is that calling `createPage` inside `onCreatePage` gives an endless cycle.
Is there any way to `createPage` quietly, without calling that callback again? Maybe there is some kind of ActionOption? ActionOptions don't seem to be documented anywhere.

Or, maybe there is some other approach I should take?
Goal is to have English page available at both `/en/blog/post` and `/blog/post/` and add some extra locale information to the page context.

**Comment by LekoArts** [comment, marked as answer]
Check if the context you're setting was already set and bail out early.

```js
exports.onCreatePage = ({ page }) => {
	if (page.context.yourKey) {
	  return
	}
}
```

**Reply by tonnyavery** [reply]
Thanks for a quick idea, works great
